### DISCUSSION TITLE
File Upload Issue

### DISCUSSION THREAD

**Discussion by Sharafudeen** [fgilde/MudBlazor.Extensions#86] [General]
Hi,

I am rendering a model that includes properties such as Name, Age, and UploadFileRequest, which is of type ICollection UploadableFile. The model meta configurations for the UploadFileRequest is just WithoutLabel setup and EditObject dialog are rendering perfectly. However, after submitting the dialog, I am unable to retrieve file information such as Data, Path, or URL. I couldn't find any samples related to this in the documentation. Could you please provide guidance? Thanks!

**Updates:**
Here is the Object Edit file upload dialog example https://www.mudex.org/dialog-object-edit **but seems it is not working as mentioned above!**

**Comment by fgilde** [comment]
Can you please check if this post helps?

https://github.com/fgilde/MudBlazor.Extensions/discussions/33

**Reply by Sharafudeen** [reply]
@fgilde Actually I am looking to retrieve the file byte array/data or the full file path after submitting the MudExDialog. Once I have this data, I can then post the file data to the API.

**Reply by fgilde** [reply]
Just get it from your model or from the EditContext.

```c#
<MudExObjectEditForm OnValidSubmit="@OnSubmit"  Value="@Model"></MudExObjectEditForm>


@code {

    public class SampleModel
    {
        [Required]
        public string Name { get; set; }
        public string Description { get; set; }

        [Required]
        [MinLength(2)]
        public UploadableFile[] Files { get; set; }
    }
    
    // Sample data for the form
    public SampleModel Model { get; set; } = new();

    private async Task OnSubmit(EditContext ctx)
    {
        var model = ctx.Model as SampleModel;
        Console.WriteLine(model.Files[0].Data.Length); // Access the file data from edit context
        Console.WriteLine(Model.Files[0].Data.Length); // Access the file data from updated the model
    }

}

```

Here you will find a sample

I created a sample for you here, also with the sample how to acess files for upload etc.

https://try.mudex.org/snippet/mEGyEAmMHGqGKEow

**Reply by Sharafudeen** [reply]
I’m not placing MudExObjectEditForm in the component instead I’m using EditObject dialog.

**Reply by fgilde** [reply]
Ok you're right now i get it. It looks like you found a bug. Damn .. I've created an [issue here](https://github.com/fgilde/MudBlazor.Extensions/issues/88) and try to fix as soon as possible

**Reply by Sharafudeen** [reply]
@fgilde Thanks Glide.. it will be very helpful..!!

**Reply by fgilde** [reply]
It' fixed in 1.7.89

https://www.nuget.org/packages/MudBlazor.Extensions/1.7.89


Now this sample works

```c#
 public class TestObject
 {
     [Required]
     public string Value { get; set; } = "Test";
     [Range(1, 100)]
     public int Number { get; set; }
     public LogLevel logLevel { get; set; }
     public SomeOther Some { get; set; }
     [Required]
     public UploadableFile UploadRequest { get; set; }
 }

 public class SomeOther
 {
     public string Id { get; set; }
     public string Name { get; set; }
 }

 private async Task ShowSampleDialog()
 {
     DialogParameters parameters = new DialogParameters
     {
         { nameof(MudExObjectEditDialog<TestObject>.AutoSaveRestoreState), true },
         { nameof(MudExObjectEditDialog<TestObject>.SetPropertiesAfterImport), true },
         { nameof(MudExObjectEditDialog<TestObject>.AllowExport), true },
         { nameof(MudExObjectEditDialog<TestObject>.AllowImport), true }
     };
     var res = await DialogService.EditObject(new TestObject(), "Auto Generated Editor for TestObject", OnSubmit, SampleDialogOptions.FullSlideFromRight, 
     meta => meta.WrapEachInMudItem(i => i.xs = 6)
         .Property(m => m.UploadRequest)
         .WithoutLabel()            
         .WithAdditionalAttributes<MudExUploadEdit<UploadableFile>>(a => a.MimeTypes = new[] { "image/*" })
         .WrapInMudItem(i => i.xs = 12), 
     parameters);

     if (!res.Cancelled)
         await JsRuntime.InvokeVoidAsync("alert", JsonConvert.SerializeObject(res.Result));
 }
 

 private async Task<string> OnSubmit(TestObject value, MudExObjectEditDialog<TestObject> dialog)
 {
     //await value.UploadRequest.EnsureDataLoadedAsync();
     Console.WriteLine(value.UploadRequest.Data.Length);
     await Task.Delay(2000); // Simulate server save
     if (value.Value == "Test")
         return "'Test' is as Value not allowed or already exists";
     if (value.Value == "Exception")
         throw new Exception("This is a test exception");
     return null;
 }
 ```

**Reply by Sharafudeen** [reply]
@fgilde Once upgraded the package from 1.7.87 to 1.7.89 I'm getting below error, (I'm using dotnet 8)

![image](https://github.com/fgilde/MudBlazor.Extensions/assets/42769344/aadd6f77-c2e2-4f3f-a5f4-80d73c3d86af)

**Reply by fgilde** [reply]
Didn't get that. Can't reproduce and there ist nothing changed. Are you sure you only have this bug with 1.7.89?

**Reply by Sharafudeen** [reply]
Nextended packages updated to latest and issues resolved.

**Reply by fgilde** [reply]
Great.

**Reply by Sharafudeen** [reply]
@fgilde File Data only getting but Path and Url are not getting while submitting the modal dialog.

![image](https://github.com/fgilde/MudBlazor.Extensions/assets/42769344/c51169be-4f91-4529-a04a-b34083496e6f)

**Reply by fgilde** [reply]
Path is only set for files that came from folder or zip upload. It's the relative path then. URL is null because it's handled as blob and parent holder is disposed. To create an url from data you can use MudExFileService or for creating data url also the static utility DataUrl

**Reply by Sharafudeen** [reply]
@fgilde Thanks a lot.
